indico/modules/categories/models/roles.py

```diff
@@ -0,0 +1,135 @@
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.util.locators import locator_property
+from indico.util.string import format_repr, return_ascii
+
+
+class CategoryRole(db.Model):
+    __tablename__ = 'roles'
+    __table_args__ = (db.CheckConstraint('code = upper(code)', 'uppercase_code'),
+                      db.Index(None, 'category_id', 'code', unique=True),
+                      {'schema': 'categories'})
+
+    is_group = False
+    is_event_role = False
+    is_category_role = True
+    is_single_person = True
+    is_network = False
+    principal_order = 2
+    principal_type = PrincipalType.category_role
+
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    category_id = db.Column(
+        db.Integer,
+        db.ForeignKey('categories.categories.id'),
+        nullable=False,
+        index=True
+    )
+    name = db.Column(
+        db.String,
+        nullable=False
+    )
+    code = db.Column(
+        db.String,
+        nullable=False
+    )
+    color = db.Column(
+        db.String,
+        nullable=False
+    )
+
+    category = db.relationship(
+        'Category',
+        lazy=True,
+        backref=db.backref(
+            'roles',
+            cascade='all, delete-orphan',
+            lazy=True
+        )
+    )
+    members = db.relationship(
+        'User',
+        secondary='categories.role_members',
+        lazy=True,
+        collection_class=set,
+        backref=db.backref('category_roles', lazy=True, collection_class=set),
+    )
+
+    # relationship backrefs:
+    # - in_attachment_acls (AttachmentPrincipal.category_role)
+    # - in_attachment_folder_acls (AttachmentFolderPrincipal.category_role)
+    # - in_category_acls (CategoryPrincipal.category_role)
+    # - in_contribution_acls (ContributionPrincipal.category_role)
+    # - in_event_acls (EventPrincipal.category_role)
+    # - in_event_settings_acls (EventSettingPrincipal.category_role)
+    # - in_session_acls (SessionPrincipal.category_role)
+    # - in_track_acls (TrackPrincipal.category_role)
+
+    def __contains__(self, user):
+        return user is not None and self in user.category_roles
+
+    @return_ascii
+    def __repr__(self):
+        return format_repr(self, 'id', 'code', _text=self.name)
+
+    @locator_property
+    def locator(self):
+        return dict(self.category.locator, role_id=self.id)
+
+    @property
+    def identifier(self):
+        return 'CategoryRole:{}'.format(self.id)
+
+    @property
+    def css(self):
+        return 'color: #{0} !important; border-color: #{0} !important'.format(self.color)
+
+    @property
+    def style(self):
+        return {'color': '#' + self.color, 'borderColor': '#' + self.color}
+
+    @staticmethod
+    def get_category_roles(cat):
+        category_roles = CategoryRole.query.join(cat.chain_query.subquery()).all()
+        return sorted(category_roles, key=attrgetter('code'))
```

Comment:
You can just add `.order_by(CategoryRole.code)` to the query:

    return CategoryRole.query....order_by(...).all()